api-server/src/roster/views.py

```diff
@@ -345,4 +353,134 @@ def get_queryset(self):
 
 class OperationListView(ListCreateAPIView):
     queryset = Operation.objects.all().order_by('-start')
-    serializer_class = OperationSerializer
\ No newline at end of file
+    serializer_class = OperationSerializer
+
+
+class AviatorLiveriesListView(ListCreateAPIView):
+    # The livery generation permissions will only be for admins
+    # Get calls will be for all logged in users and return the files from blob
+    permission_classes = [permissions.IsAuthenticated]
+    azure_key = os.getenv('AZURE_STORAGE_KEY')
+    account_name = 'jtf191blobstorage'
+    azure_container = 'static'
+
+    block_blob_service = BlockBlobService(account_name=account_name, account_key=azure_key)
+
+    def get(self, request):
+        blobs = self.block_blob_service.list_blobs(container_name=self.azure_container, prefix="livery/")
+        archive = BytesIO()
+        with zipfile.ZipFile(archive, 'w') as zip_file:
+            for blob in blobs:
+                file = self.block_blob_service.get_blob_to_bytes(
+                        container_name=self.azure_container,
+                        blob_name=blob.name)
+                zip_file.writestr(blob.name, file.content)
+
+        response = HttpResponse(archive.getvalue(), content_type='application/force-download')
+        response['Content-Disposition'] = 'attachment; filename="%s"' % 'liveries.zip'
+        return response
+
+    def post(self, request):
+        """
+        Steps:
+        - Get all aviators
+        - Per aviator:
+            - Get Position
+            - Get Squadron
+            - Per Airframe
+                - Get liveries from Blob based on the above props
+                - Get Skin details from Django
+                - Create DDS and LUA files
+                - Upload those to Blob
+        """
+        if not request.user.is_staff:
+            return Response({'detail': ['Not allowed']},
+                            status=status.HTTP_403_FORBIDDEN)
+
+        aviators = Aviator.objects.all()
+        if not aviators:
+            return Response({'detail': ['No Aviators found']},
+                            status=status.HTTP_404_NOT_FOUND)
+
+        for aviator in aviators:
+            squadron = aviator.squadron.designation
+
+            # This is needed as DCS needs a very specific name for each airframe
+            airframe_label = aviator.squadron.air_frame.dcs_livery_label
+
+            # Get liveries from DB
+            squadron_livery = Livery.objects.filter(squadron__designation = squadron)\
+                    .filter(squadron__air_frame__name = aviator.squadron.air_frame.name)\
+                    .filter(position_code = aviator.position_code)
```

Comment:
```suggestion

             # This is needed as DCS needs a very specific name for each airframe
            airframe_label = aviator.squadron.air_frame.dcs_livery_label

            # Get liveries from DB
            squadron_livery = Livery.objects.filter(squadron=aviator.squadron, 
                                                                             position_code=aviator.position_code)
```

this is a little more concise

Reply:
👍